src/test/java/uk/gov/pay/directdebit/pact/PublicApiContractTest.java

```diff
@@ -92,10 +94,13 @@ public void aGatewayAccountWithExternalIdAndAConfirmedMandateExists(Map<String,
 
     @State("three payment records exist")
     public void threePaymentRecordsExist(Map<String, String> params) {
-        testGatewayAccount.withExternalId(params.get("gateway_account_id")).insert(app.getTestContext().getJdbi());
+        testGatewayAccount.withPaymentProvider(PaymentProvider.GOCARDLESS)
+                .withExternalId(params.get("gateway_account_id"))
+                .insert(app.getTestContext().getJdbi());
         MandateFixture testMandate = MandateFixture.aMandateFixture()
                 .withGatewayAccountFixture(testGatewayAccount)
-                .withExternalId(MandateExternalId.valueOf(params.get("agreement_id")));
+                .withPaymentProviderId(GoCardlessMandateId.valueOf("MD1"))
```

Comment:
Should this be on params?

Reply:
no because the only thing we care about when searching payments for this pact test is that a gateway account id with x exists and three payments with mandate id y exists.

Reply:
setting the MD1 here is just to ensure a provider_id is returned on a payment when searching